test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Commands/PushCommandTest.cs

```diff
@@ -245,6 +256,512 @@ public void PushCommand_Server_SkipDuplicate_IsSpecified_PushProceeds()
+                string snupkgFullPath2 = Path.Combine(packageDirectory, snupkgFileName2);
+                //Create another snupkg in test directory.
+                WriteSnupkgFile(snupkgFullPath2);
+
+                string wildcardPush = "*.snupkg";
+
+                using (var server = CreateAndStartMockV3Server(packageDirectory, out string sourceName))
+                {
+                    SetupMockServerAlwaysDuplicate(server);
+
+                    // Act
+
+                    //Since this is V3, this will trigger 2 pushes: one for nupkgs, and one for snupkgs.
+                    var result = CommandRunner.Run(
+                        nuget,
+                        packageDirectory,
+                        $"push {wildcardPush} -Source {sourceName} -Timeout 110 -SkipDuplicate",
+                        waitForExit: true,
+                        timeOutInMilliseconds: 120000); // 120 seconds
+
+                    // Assert
+
+                    //Ignoring filename in File Not Found error since the error should not appear in any case.
+                    string genericFileNotFoundError = string.Format(MESSAGE_FILE_DOES_NOT_EXIST, string.Empty);
+
+                    //Nupkg should be an ignored conflict, so its snupkg should push.
+                    //TODO: Once SkipDuplicate is passed-through to the inherit snupkg push, the following TODO's should be corrected.
+                    Assert.False(result.Success, "Expected to fail to push a snupkg when the nupkg is a duplicate."); //TODO: Should succeed since duplicates are skipped.
+                    Assert.Contains(MESSAGE_RESPONSE_NO_SUCCESS, result.Errors); //TODO: snupkg duplicate is ignored
+                    Assert.DoesNotContain(MESSAGE_PACKAGE_PUSHED, result.Output); //snupkgFileName and snupkgFileName2 pushed
+                    Assert.DoesNotContain(MESSAGE_EXISTING_PACKAGE, result.Errors); //TODO: Expect this message once SkipDuplicate is working.
+                    Assert.DoesNotContain(genericFileNotFoundError, result.Errors);
+
+                    Assert.Contains(snupkgFileName, result.AllOutput); //first snupkg is attempted as push
+                    Assert.DoesNotContain(snupkgFileName2, result.AllOutput); //TODO: second snupkg should be attempted.
+
+                    Assert.DoesNotContain(nupkgFileName, result.AllOutput); //nupkgs should not be attempted in push
+                    Assert.DoesNotContain(nupkgFileName2, result.AllOutput); //nupkgs should not be attempted in push
+                }
+            }
+        }
+
+
+        /// <summary>
+        /// When pushing snupkg with SkipDuplicate, a 409 Conflict is ignored and any message from the server is shown appropriately.
+        /// TODO: bug fixes will come from https://github.com/NuGet/Home/issues/8148
+        /// </summary>
+        [Fact]
+        public void PushCommand_Server_Snupkg_FilenameSnupkgExists_SkipDuplicate_ServerMessage()
```

Comment:
It's not clear to me from the speclet (it'd be nice to put the spec in the designs repo for future tracking) whether the nuget.org behavior is changing, or all of it is client side.

Reply:
This test has nothing to do with NuGet.org. These are NuGet Client Functional Tests, which are testing how the Client handles file not found, and pushing snupkgs with corresponding nupkgs.

Reply:
I couldn't follow what the behavior should be based on the spec.
You cleared it up somewhere in the other comments.